formats/hocon/src/test/kotlin/kotlinx/serialization/hocon/HoconEncoderTest.kt

```diff
@@ -0,0 +1,130 @@
+package kotlinx.serialization.hocon
+
+import kotlinx.serialization.*
+import org.junit.*
+
+class HoconEncoderTest {
```

Comment:
Sorry if I missed that, but I don't see the test that checks that serializing an Int, List, etc. leads to a `SerializationException("Value of type '${configValue.valueType()}' can't be used at the root of HOCON Config.")`. Can you add it, please? And for other exception cases too, if necessary.

Reply:
Ok, I'll add it

Reply:
It seems that work is almost finished; can you add this test (and resolve conflicts with https://github.com/Kotlin/kotlinx.serialization/pull/1795) so we can include your work in the release planned for this week?

Reply:
Ok. I'l try to finish it ASAP

Reply:
Added tests for unsupported root objects and map keys in [two](https://github.com/Kotlin/kotlinx.serialization/pull/1740/commits/fe4c06760b7c0abfd8c2a714e7cd17ae3b30e41b) [last](https://github.com/Kotlin/kotlinx.serialization/pull/1740/commits/c4edfe91551375e3cef38e0a329ed99dcde72ce8) commits.